### PR TITLE
`j = <expr>:<expr>` is now with=TRUE with at least one call expr

### LINKED ISSUE #2069: Confusing behavior with DT[, min(var):max(var)]
With this data...

    library(data.table)

    DT = data.table(
        id = c(1L, 1L, 1L, 2L, 2L, 2L, 3L, 3L, 3L, 4L, 4L, 4L, 5L, 5L, 5L, 6L, 6L, 6L), 
        t = c(4L, 9L, 14L, 1L, 4L, 9L, 5L, 14L, 18L, 1L, 3L, 12L, 3L, 5L, 7L, 4L, 10L, 13L)
    )

        id  t
     1:  1  4
     2:  1  9
     3:  1 14
     4:  2  1
     5:  2  4
     6:  2  9
     7:  3  5
     8:  3 14
     9:  3 18
    10:  4  1
    11:  4  3
    12:  4 12
    13:  5  3
    14:  5  5
    15:  5  7
    16:  6  4
    17:  6 10
    18:  6 13

I expected `DT[, min(t):max(t)]` to be the same as `with(DT, min(t):max(t))` -- just a vector of integers counting over that interval. I got something quite different:

    >     DT[, min(id):max(id)]
        id
     1:  1
     2:  1
     3:  1
     4:  2
     5:  2
     6:  2
     7:  3
     8:  3
     9:  3
    10:  4
    11:  4
    12:  4
    13:  5
    14:  5
    15:  5
    16:  6
    17:  6
    18:  6
    >     DT[, seq(min(id), max(id))]
    [1] 1 2 3 4 5 6
    >     
    > 
    >     DT[, min(t):max(t)]
         t
     1:  4
     2:  9
     3: 14
     4:  1
     5:  4
     6:  9
     7:  5
     8: 14
     9: 18
    10:  1
    11:  3
    12: 12
    13:  3
    14:  5
    15:  7
    16:  4
    17: 10
    18: 13
    >     DT[, seq(min(t), max(t))]
     [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17

Maybe this is a side-effect of the move to make `DT[, j]` behave more like `DF[, j]`? I didn't check to see if it also happened in older versions.

### PR DESCRIPTION
**Pull request by jangorecki**
closes #2069

### CODE REVIEW

**Comment by MichaelChirico** [comment]
I think the PR title is misleading, it looks like 'symbol:symbol' but the PR is actually about 'call():call()', right?

File: NEWS.md [hunk 1]
```diff
@@ -109,6 +109,8 @@ unit = "s")
 
 13. A relatively rare case of segfault when combining non-equi joins with `by=.EACHI` is now fixed, closes [#4388](https://github.com/Rdatatable/data.table/issues/4388).
 
+14. When using `:` operator on a _call_ object(s) inside `j` argument, it could have been improperly recognised as an attempt to select columns (`with=FALSE`) rather than evaluate expression within a data.table, [#2069](https://github.com/Rdatatable/data.table/issues/2069). Thanks to @franknarf1 for reporting.
```
Comment (MichaelChirico) [inline NEWS.md:112]: Please rephrase the NEWS item, I doubt it will be well understood. I would lead with a concrete example and then explain the behavior.

File: R/data.table.R [hunk 2]
```diff
@@ -253,18 +253,18 @@ replace_dot_alias = function(e) {
...
+          root %chin% c("","c","paste","paste0","-","!") &&                                     ## x[, c("V1","V2")]; x[, paste("V",1:2,sep="")]; x[, paste0("V",1:2)]
           missingby )) {   # test 763. TODO: likely that !missingby iff with==TRUE (so, with can be removed)
       # When no variable names (i.e. symbols) occur in j, scope doesn't matter because there are no symbols to find.
       # If variable names do occur, but they are all prefixed with .., then that means look up in calling scope.
       # Automatically set with=FALSE in this case so that DT[,1], DT[,2:3], DT[,"someCol"] and DT[,c("colB","colD")]
       # work as expected.  As before, a vector will never be returned, but a single column data.table
       # for type consistency with >1 cases. To return a single vector use DT[["someCol"]] or DT[[3]].
       # The root==":" is to allow DT[,colC:colH] even though that contains two variable names.
-      # root == "-" or "!" is for tests 1504.11 and 1504.13 (a : with a ! or - modifier root)
+      # root == "-" or "!" is for tests 1504.10 and 1504.12 (a : with a ! or - modifier root)
```
Comment (MichaelChirico) [inline R/data.table.R:267]: is this change intended?

**Comment by MichaelChirico** [comment]
This will be a breaking change, `DT[, (a):(b)]` worked fine previously as being identical to `a:b` selecting cols `a` through `b`.

Better maybe to ensure it's an "actual" call like `foo()` instead of just `(`.

c.f.

```r
DT=as.data.table(mtcars)
DT[, (hp):(wt)]
```

File: R/data.table.R (hunk 2 above)
Comment (jangorecki) [reply R/data.table.R:267]: Doesn't seem to

**Comment by MichaelChirico** [comment]
I got the "refuse to merge due to unrelated histories" issue again when trying to merge, hence force-pushed the small change split directly from current `master`. Not much history was lost. At least this time I didn't first close+delete the GH branch which caused more chaos.

**Comment by MichaelChirico** [comment]
> This will be a breaking change, `DT[, (a):(b)]` worked fine previously as being identical to `a:b` selecting cols `a` through `b`.
> 
> Better maybe to ensure it's an "actual" call like `foo()` instead of just `(`.
> 
> c.f.
> 
> ```r
> DT=as.data.table(mtcars)
> DT[, (hp):(wt)]
> ```

Let's see what revdeps say. I _do_ see some usage like this, but glancing through the search results I don't see any data.table `[` usage, only matrix:

https://github.com/search?q=lang%3AR+%2F%5B%2C%5D%5Cs*%5B%28%5D%5Ba-z._0-9%5D%2B%5B%29%5D%5B%3A%5D%5B%28%5D%5Ba-z0-9._%5D%2B%5B%29%5D%2F&type=code

**Comment by MichaelChirico** [comment]
Thanks @jangorecki!
